### PR TITLE
fix: @astrojs/telemetry crashing astro dev/build when using optional integrations 

### PR DESCRIPTION
**Pull request by okikio**
## Changes

- Fixes @astrojs/telemetry crashing astro dev/build when using optional integrations 

## Testing

- Added tests for filtering out falsy integrations in `@astrojs/telemetry` package
- Added tests to verify the count of optional integrations in use is accurate

## Docs

No, docs needed, this is a bug fix. 

cc @withastro/maintainers-core 
>  ~~^ You may want to pay attention to this change as the data you receive from telemetry won't include the usage data of optional integrations~~

I've setup a count of the number of optional integrations in use, as well as the total number of optional integrations in use, this should at least help identify just how many projects use optional integrations

### CODE REVIEW

File: packages/telemetry/src/events/session.ts [hunk 1]
```diff
@@ -89,14 +91,23 @@ export function eventCliSession(
...
 	const configValues = userConfig
 		? {
 				markdownPlugins: [
 					userConfig?.markdown?.remarkPlugins ?? [],
 					userConfig?.markdown?.rehypePlugins ?? [],
 				].flat(1),
 				adapter: userConfig?.adapter?.name ?? null,
-				integrations: userConfig?.integrations?.map((i: any) => i.name) ?? [],
+				// Filter out falsy integrations
+				integrations: userConfig?.integrations?.filter((i: any) => {
```
Comment (tony-sull) [inline packages/telemetry/src/events/session.ts:103]: Mostly just personal preference, but I try to avoid side effects inside of `filter` functions.  Thoughts on moving the filter up with something like this?

```
const integrations = userConfig?.integrations?.filter(Boolean) ?? [];

const configValues = userConfig
  ? {
    // ...
    integrations,
    totalIntegrations: integrations?.length,
    optionalIntegrations: userConfig?.integrations?.length - integrations?.length
  }
```

File: packages/telemetry/src/events/session.ts [hunk 2]
```diff
@@ -36,6 +36,8 @@ interface EventCliSessionInternal extends EventCliSession {
 	config?: ConfigInfo;
 	configKeys?: string[];
 	flags?: string[];
+	optionalIntegrations?: number;
+	totalNumOfIntegrations?: number;
```
Comment (tony-sull) [inline packages/telemetry/src/events/session.ts:40]: <insert bike shed joke>

nit: Might be nice to have the names look similar, maybe just `totalIntegrations` or `optionalIntegrationsCount` and `integrationsCount`?

File: packages/webapi/mod.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
+export { AbortController, AbortSignal, alert, atob, Blob, btoa, ByteLengthQueuingStrategy, cancelAnimationFrame, cancelIdleCallback, CanvasRenderingContext2D, CharacterData, clearTimeout, Comment, CountQueuingStrategy, CSSStyleSheet, CustomElementRegistry, CustomEvent, Document, DocumentFragment, DOMException, Element, Event, EventTarget, fetch, File, FormData, Headers, HTMLBodyElement, HTMLCanvasElement, HTMLDivElement, HTMLDocument, HTMLElement, HTMLHeadElement, HTMLHtmlElement, HTMLImageElement, HTMLSpanElement, HTMLStyleElement, HTMLTemplateElement, HTMLUnknownElement, Image, ImageData, IntersectionObserver, MediaQueryList, MutationObserver, Node, NodeFilter, NodeIterator, OffscreenCanvas, ReadableByteStreamController, ReadableStream, ReadableStreamBYOBReader, ReadableStreamBYOBRequest, ReadableStreamDefaultController, ReadableStreamDefaultReader, Request, requestAnimationFrame, requestIdleCallback, ResizeObserver, Response, setTimeout, ShadowRoot, structuredClone, StyleSheet, Text, TransformStream, TreeWalker, URLPattern, Window, WritableStream, WritableStreamDefaultController, WritableStreamDefaultWriter, } from './mod.js';
```
Comment (tony-sull) [inline packages/webapi/mod.d.ts:2]: Think this is just a whitespace change from the compiler, probably worth reverting it since its unrelated

File: packages/telemetry/test/session-event.test.js [hunk 4]
```diff
@@ -411,6 +411,29 @@ describe('Session event', () => {
 		});
 	});
 
+	describe('config.integrations + optionalIntegrations + totalNumOfIntegrations', () => {
```
Comment (tony-sull) [inline packages/telemetry/test/session-event.test.js:414]: Test coverage, thank you!!  🎉 🥳 

Would you mind adding a case where all the integrations are falsy?  Feels like one of those edge cases that I could break without realizing it 😅

**Review by tony-sull — CHANGES_REQUESTED** [review]
Excellent, thanks for filing the PR so quickly!  Added a few small comments, but nothing big 👍

**Comment by okikio** [comment]
@tony-sull I've resolved all the comments, PTAL

File: packages/telemetry/src/events/session.ts [hunk 5]
```diff
@@ -125,6 +129,10 @@ export function eventCliSession(
 		// Config Values
 		config: configValues,
 		flags: cliFlags,
+		// Total Number of Integrations
+		totalIntegrations: userConfig?.integrations?.length,
```
Comment (matthewp) [inline packages/telemetry/src/events/session.ts:133]: We don't need this field as Mixpanel can give us this in its reporting. We've avoided during this sort of pre-processing on purpose.

**Comment by matthewp** [comment]
Can we remove the total fields? We can debate their value in another PR or on Discord, I'm not completely against them but we have avoided adding those types of things on purpose so far. But I think fixing the bug is the important thing to get out in this PR.

File: packages/telemetry/src/events/session.ts [hunk 6]
```diff
@@ -125,6 +129,10 @@ export function eventCliSession(
...
+		// Optional integrations
+		optionalIntegrations: userConfig?.integrations?.length - integrations?.length
```
[the remark this answers is not in the document]
Comment (okikio) [reply packages/telemetry/src/events/session.ts:135]: I'll create a new pr with the total integrations and optional integrations fields

[the remark this answers is not in the document]
Comment (okikio) [reply packages/telemetry/src/events/session.ts]: Here is the new pr #3614 with the optional integrations field. We can continue our discussion on this feature there.

**Review by natemoo-re — APPROVED** [review]
Thanks for splitting out the PRs for further discussion. LGTM!

**Review by tony-sull — APPROVED** [review]
Sorry, I saw 3614 merged and didn't notice it was split out from this original PR!

@okikio looks like there's just a quick merge conflict to take care of then we can merge this one in too 👍

**Comment by okikio** [comment]
@tony-sull When I split the pr, I made sure #3614 would also fix this issue, since #3614 has already been merged this pr in no longer necessary.

**Comment by tony-sull** [comment]
Oh nice!  That makes me feel a better, I looked at the #3614 changes and could have sworn the null checks were in there 😄
